protocol/x/subaccounts/keeper/isolated_subaccount.go

```diff
@@ -2,16 +2,49 @@ package keeper
 
 import (
 	"math"
+	"math/big"
 
 	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	assettypes "github.com/dydxprotocol/v4-chain/protocol/x/assets/types"
 	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
 	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 )
 
+type positionStateTransition uint
+
+const (
+	opened positionStateTransition = iota
+	closed
+)
+
+var positionStateTransitionStringMap = map[positionStateTransition]string{
+	opened: "opened",
+	closed: "closed",
+}
+
+func (t positionStateTransition) String() string {
+	result, exists := positionStateTransitionStringMap[t]
+	if !exists {
+		return "UnexpectedStateTransitionError"
+	}
+
+	return result
+}
+
+// Represents a state transition for an isolated perpetual.
```

Comment:
Nit: `state transition for an isolated perpetual position`?

Reply:
Updated naming of the type as well.